cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java

```diff
@@ -107,6 +110,30 @@ public void testGenerationWithExternalIncrements() throws Exception {
         assertThat(triggersIds.size()).isEqualTo(2);
     }
 
+    /**
+     * Tests generation of external increments where its trigger does not match
+     * @throws IOException
+     *             test fails
+     */
+    public void testGenerationWithExternalIncrementsFailsWhenExternalTriggerNotMatch() throws IOException {
+        // given
+        Object input = PluginMockFactory.createSimpleJavaConfigurationMock();
+
+        File folder = tmpFolder.newFolder("GenerationTest");
+        File target = new File(folder, "generated.txt");
+        FileUtils.write(target, "base");
+
+        // when
+        CobiGen cobigen =
+            CobiGenFactory.create(new File(testFileRootPath + "externalIncrementsGenerationException").toURI());
+
+        // exception is thrown while getting all increments
+        assertThatThrownBy(() -> {
+            cobigen.getMatchingIncrements(input);
+        }).isInstanceOf(InvalidConfigurationException.class).hasMessageContaining(
+            "An external incrementRef to valid_increment_composition::0 is referenced from external_incrementref but its trigger does not match");
+    }
+
```

Comment:
Could you add also another positive test, which shows that an external template is generated?

Reply:
The positive test is already written by @jdiazgon  in GenerationTest.testGenerationWithExternalIncrements

Reply:
ah but I thought external templates are not yet generated, right? Why could the positive test already exist?

Reply:
@maybeec the external templates are always generated if the external trigger is matched. Yeah, a test exists :) Please refer to the class GenerationTest